gtfu/src/main/java/gtfu/Stats.java

```diff
@@ -0,0 +1,51 @@
+package gtfu;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collection;
+
+public class Stats {
+
+    private double avg;
+    private double min = Double.MAX_VALUE;
+    private double max = Double.MIN_VALUE;
+
+
+    public Stats(Collection<? extends StatValue> c) {
+        double total = 0;
+        int count = 0;
+
+        for (StatValue v : c) {
+            Double value = v.getValue();
+            if (value == null) continue;
+            count++;
+            total += value;
+
+            if (value < min) {
+                min = value;
+            }
+
+            if (value > max) {
+                max = value;
+            }
+        }
+
+        avg = total / count;
```

Comment:
Finding a bug in my own code. This should be:
`avg = count == 0 ? 0 : total / count;`
to avoid an exception if the collection is empty or all calls to `getValue()` return null

Reply:
Actually, avg gets initialized to `0` by the VM, so to make things a little less cryptic, you can just say:

`if (count > 0) avg = total / count;`